csm/conf/uds.py

```diff
@@ -80,7 +80,7 @@ def generate_haproxy_frontend_config():
     bind 127.0.0.1:5000
     bind ::1:5000
     bind {cluster_ip}:5000
-    bind {mgmt_vip}:5000
+    bind {mgmt_vip}:5000 transparent
```

Comment:
For transparent binding, there are some prerequisites. How are we going to ensure that these prerequisites are met?
https://www.haproxy.com/blog/howto-transparent-proxying-and-binding-with-haproxy-and-aloha-load-balancer/

Reply:
> For transparent binding, there are some prerequisites. How are we going to ensure that these prerequisites are met?
https://www.haproxy.com/blog/howto-transparent-proxying-and-binding-with-haproxy-and-aloha-load-balancer/

We do not require the passive node to forward traffic to the active node in that scenario, because ARP announcements will contain information on the active node only, so the bulk of these configuration changes are not needed, only those required by HAproxy so it does not spit out an error.

> Also, looks like transparent binding has different purpose, not to avoid binding if ip address is not available.

You are right, this approach does not prevent HAproxy from binding to that IP address. It just relies on the way transparent binding works to avoid the breaking failure.

To be honest the option for transparent binding was a split-second decision but I think it might prove useful. Currently:

* HAproxy binds both to cluster IP and mgmt VIP (transparently)
* UDS binds to private data IP
* CSM/USL config determines if USL will advertise cluster IP or mgmt VIP.

The "beauty" of this approach brings the number of moving parts on the solution down to one: we just need to configure which IP address will be advertised by CSM/USL. The rest is all the same: UDS and HAproxy configuration entries are the same on both scenarios, for data transfers on data network or on management network.

Another possibility (which I thought of later) is:

- For data transfers on management network
  * No UDS-related HAproxy config entries
  * UDS config entry binds to 0.0.0.0 (required because USL talks to it via `localhost`)
  * CSM/USL config maps to mgmt VIP
- For data transfers on data network
  * UDS-related HAproxy config entry binds to `localhost` and cluster IP
  * UDS config entry binds to private data network
  * CSM/USL config maps to cluster IP